File: src/main/java/net/kigawa/util/plugin/stage/StageData.java
package net.kigawa.util.plugin.stage;

import net.kigawa.util.plugin.recorder.RecorderData;

public class StageData extends RecorderData {

    public int[] getStartLoc() {
        return startLoc;
    }

    public void setStartLoc(int[] startLoc) {
        this.startLoc = startLoc;
    }

    int[] startLoc = new int[3];

    public int[] getStageLoc() {
        return stageLoc;
    }

    public void setStageLoc(int[] stageLoc) {
        this.stageLoc = stageLoc;
    }

    int[] stageLoc = new int[6];
    String stageWorld;

    public String getStageWorld() {
        return stageWorld;
    }

    public void setStageWorld(String stageWorld) {
        this.stageWorld = stageWorld;
    }
}

File: src/main/java/net/kigawa/onigoplugin/util/plugin/game/stage/StageData.kt
package net.kigawa.onigoplugin.util.plugin.game.stage

import net.kigawa.onigoplugin.util.plugin.all.recorder.RecorderData
import org.bukkit.Location

class StageData(name: String = "") : RecorderData(name) {
  var stageWorld: String? = null
  var startLoc = IntArray(3)
  var stageLoc = IntArray(6)
  fun setStageStartLoc(location: Location) {
    stageLoc[0] = location.blockX
    stageLoc[1] = location.blockY
    stageLoc[2] = location.blockZ
  }

  fun setStageEndLoc(location: Location) {
    stageLoc[3] = location.blockX
    stageLoc[4] = location.blockY
    stageLoc[5] = location.blockZ
  }

  fun setStartLoc(location: Location) {
    startLoc[0] = location.blockX
    startLoc[1] = location.blockY
    startLoc[2] = location.blockZ
  }
}
